h5netcdf/tests/test_h5netcdf.py

```diff
@@ -84,6 +89,29 @@ def get_hdf5_module(resource):
         return h5py
 
 
+def h5dump(fn: str):
+    """Call h5dump on an h5netcdf file."""
+    import re
+    import subprocess
+
+    out = subprocess.run(
+        ["h5dump", "-A", fn], check=False, capture_output=True
+    ).stdout.decode()
+
+    # Strip non-deterministic components
+    out = re.sub(r'DATASET [0-9]+ "', 'DATASET XXXX "', out)
+
+    # Strip the _NCProperties header, which includes software versions which won't match.
+    out = re.sub(
+        r'ATTRIBUTE "_NCProperties" {.*}',
```

Comment:
Add `?` to regex, otherwise this will swallow the whole file, ending up with { ... }.
```suggestion
        r'ATTRIBUTE "_NCProperties" {.*?}',
```

Reply:
Sorry about the regexp mess, I should have suspected it was too good to be true. I can take a look at the regexp itself and add a test for it.

Reply:
The DATA regexp doesn't work for me locally, stripping out the (1):

netcdf4-python
```
'         DATA {\n'
'         (0): (DATASET XXXX "/n1"), (DATASET XXXX "/n2"),\n'
'         (2): (DATASET XXXX "/nchar")\n'
```

legacyapi
```
'         DATA {\n'
'         (0): (DATASET XXXX "/n1"),\n'
'         (1): (DATASET XXXX "/n2"),\n'
'         (2): (DATASET XXXX "/nchar")\n'
```

Any idea why this might happen?

Reply:
This is fixed now.